File: BasicToAdvance/Python/Day-12/BinarySearch.py
nums = [12,15,18,20,30,40]

target = 30

def binarySearch(nums,target):
    n = len(nums)
    high = n-1
    low = 0

    while low<=high:
        mid = (low + high)//2

        if nums[mid] == target:
            return mid
        elif nums[mid] < target:
            low = mid + 1
        else:
            high = mid - 1
    return -1

x = binarySearch(nums,target)
print(x)

File: BasicToAdvance/Java/Day-12/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] arr = {12,18,26,30,45,62};
        int target = 30;

        int x = binarySearch(arr,target);
        System.err.println(x);
    }

    public static int binarySearch(int[] arr, int target){
        int n = arr.length;
        int high = n-1;
        int low = 0;

        while (low<=high) {
            int mid = (low + high)/2;
            if (arr[mid]==target) {
                return mid;
            }else if (arr[mid]<target) {
                low = mid + 1;
            }else{
                high = mid - 1;
            }
        }
        return -1;
    }
}
